react-native/ios/RealmReact/RealmReact.mm

```diff
@@ -55,20 +56,21 @@ - (JSGlobalContextRef)ctx;
     id rctJSContext = object_getIvar(executor, contextIvar);
     if (!rctJSContext && create) {
         Class RCTJavaScriptContext = NSClassFromString(@"RCTJavaScriptContext");
-        NSMethodSignature *signature = [RCTJavaScriptContext instanceMethodSignatureForSelector:@selector(initWithJSContext:)];
-        assert(signature);
 
-        // React Native 0.14+ expects a JSContext here, but we also support 0.13.x, which takes a JSGlobalContextRef.
-        if (!strcmp([signature getArgumentTypeAtIndex:2], "@")) {
-            JSContext *context = [[JSContext alloc] init];
-            rctJSContext = [[RCTJavaScriptContext alloc] initWithJSContext:(void *)context];
+        NSMethodSignature *signature = [RCTJavaScriptContext instanceMethodSignatureForSelector:@selector(initWithJSContext:onThread:)];
+        if (signature) {
+            // for RN 0.28.0+
+            rctJSContext = [executor context];
```

Comment:
Just so you know, the `RCTJavaScriptContextCreatedNotification` had been reliably posted right after creating the `JSContext` since React Native v0.26.0 (see [e195bf9](https://github.com/facebook/react-native/commit/e195bf9495c5f2cb4d35b2b9f777dcd2464ec5b9)). Also, a "reliable hack" would be to swizzle `-[JSContext initWithVirtualMachine:]`. :smile: